tools/core/llm.py

```
@@ -0,0 +1,78 @@
+                "messages": messages,
+                "temperature": temperature,
+            }
+        ).encode("utf-8"),
+        headers={
+            "Authorization": f"Bearer {api_key()}",
+            "Content-Type": "application/json",
+            "HTTP-Referer": "https://github.com/nica-ev/circuswiki",
+            "X-Title": title,
+        },
+        method="POST",
+    )
```

Ruff (0.15.15)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)